File: 1448-maximum-69-number/1448-maximum-69-number.c
#include<stdio.h>
#include<string.h>

int maximum69Number (int num) {
    
    char answer[5];
    char temp[5];
    sprintf(temp, "%d", num);
    bool flag = true;
    for (int i = 0; i < strlen(temp); ++i) {
        if (temp[i] == '6' && flag) {
            answer[i] = '9';
            flag = false;
        } else {
            answer[i] = temp[i];
        }
    }
    int answer_int = atoi(answer);

    return answer_int;


}

File: 1448-maximum-69-number/1448-maximum-69-number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        answer = ""
        flag_i = -1

        for i, c in enumerate(str(num)):
            if c == "6":
                answer += "9"
                flag_i = i
                break
            answer += c
        
        if flag_i != -1:
            answer += str(num)[flag_i + 1:]
        return int(answer)